blockchain/state/state_object.go

```diff
@@ -304,7 +304,7 @@ func (s *stateObject) SetStorage(storage map[common.Hash]common.Hash) {
 // IsContractAccount returns true is the account has a non-empty codeHash.
 func (s *stateObject) IsContractAccount() bool {
 	acc := account.GetProgramAccount(s.account)
-	if acc != nil && !bytes.Equal(acc.GetCodeHash(), emptyCodeHash) {
+	if acc != nil && acc.Type() == account.SmartContractAccountType {
```

Comment:
Either change the comment or logic.
- IsContractAccount returns true if the account is of SCA (0x02) type
- IsContractAccount returns true if the account has nonempty codehash.
See the usages and try to determine the original intention.

Reply:
> IsContractAccount returns true if the account has nonempty codehash.

I agree with this option.
The comments indicate specific enough behavior that we should consider compatibility for this.

Reply:
This function is currently used in the next API which checks if it is a ContractAccount, and checking the documentation it seems correct to return false for an empty codeHash, so we will revert the condition regarding the codeHash, thanks.